runelite-client/src/main/java/net/runelite/client/plugins/regenmeter/RegenMeterPlugin.java

```diff
@@ -185,8 +202,38 @@ else if (currentHP > maxHP)
 		{
 			notifier.notify("Your next hitpoint will regenerate soon!");
 		}
+
+		soulStackPercentage = ticksSinceSoulStackDecay / (double) SOUL_STACK_DECAY;
+		if (client.getVarpValue(VarPlayer.SOUL_STACK) == 0)
+		{
+			//No Decay when at 0 hits
+			ticksSinceSoulStackDecay = 0;
+		}
+		else
+		{
+			ticksSinceSoulStackDecay = (ticksSinceSoulStackDecay + 1) % SOUL_STACK_DECAY;
+			soulStackPercentage = 1 - soulStackPercentage;
+		}
 	}
 
+	@Subscribe
+	public void onAnimationChanged(final AnimationChanged event)
+	{
+		Player local = client.getLocalPlayer();
+
+		if (event.getActor() != local)
+		{
+			return;
+		}
+
+		int animId = local.getAnimation();
+		if (animId == 10172 && client.getVarpValue(VarPlayer.SOUL_STACK) == 5)
```

Comment:
This animation ID should go into the AnimationID api class.

Reply:
added this